packages/editor/src/components/collab-sidebar/hooks.js

```diff
@@ -29,10 +29,13 @@ export function useBlockComments( postId ) {
 			const commentId =
 				getBlockAttributes( clientId )?.metadata?.commentId;
 			if ( commentId ) {
-				results[ commentId ] = clientId;
+				results.push( {
+					blockId: clientId,
+					commentId,
+				} );
```

Comment:
We should not do this as it can affect the performance. The array/object shape was intentional. Related: https://github.com/WordPress/gutenberg/pull/71869#discussion_r2375349748.

Can we do `results[ clientId ] = commentId`, if we want to match block order instead of comment order?

P.S. If you're running a local install with `SCRIPT_DEBUG` enabled, you should see the warning in the DevTools console.

Reply:
Just realized how to fix this. Here's the PR - #72112.